src/main/java/ch/uzh/ifi/hase/soprafs23/controller/GroupController.java

```diff
@@ -328,13 +326,18 @@ public List<IngredientGetDTO> getFinalIngredientsOfGroupById(@PathVariable Long
             throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, String.format("You are not a member of the group with id %d.", groupId));
         }
 
-        // 409 - One of the members hasn't voted yet
-        for (Long memberId : memberIds) {
-            User member = userService.getUserById(memberId);
-            if (member.getVotingStatus() != UserVotingStatus.VOTED) {
-                throw new ResponseStatusException(HttpStatus.CONFLICT, "Not all members of the group have voted yet.");
+        // Check that group state is FINAL / that all members have voted
+        GroupState groupState = group.getGroupState();
+        if(!groupState.equals(GroupState.FINAL)) {
+            for (Long memberId : memberIds) {
+                User member = userService.getUserById(memberId);
+                if (member.getVotingStatus() != UserVotingStatus.VOTED) {
+                    throw new ResponseStatusException(HttpStatus.CONFLICT, "Not all members of the group have voted yet."); // 409
```

Comment:
as discussed I Would recommend to turn it into a 202 "Accepted - The request has been accepted for processing, but the processing has not been completed." This way its a similar approach for frontend, but not throwing frightening errors which we thought of dealing with anyway :D